cs-minibot-platform-src/src/main/java/minibot/BaseHTTPInterface.java

```diff
@@ -96,48 +101,151 @@ public static void main(String[] args) {
+                    int size = je.getAsJsonObject().get("size").getAsInt();
+                    PhysicalObject po = new PhysicalObject(name, 100,
+                            simulator.getWorld(), (float)position[0],
+                            (float)position[1], size, angle);
+                    simulator.importPhysicalObject(po);
+                }
+                //for bots listed in scenario
+                else {
+                    Bot newBot;
+                    name = "Simbot"+name;
+                    SimBotConnection sbc = new SimBotConnection();
+                    SimBot simbot;
+                    simbot = new SimBot(sbc, name, 50, simulator.getWorld(), 0.0f,
+                            0.0f, (float) position[0], (float)
+                            position[1], angle, true);
+                    newBot = simbot;
+
+                    simulator.importPhysicalObject(simbot.getMyPhysicalObject());
+
+                    // Color sensor TODO put somewhere nice
+//                    ColorIntensitySensor colorSensorL = new ColorIntensitySensor((SimBotSensorCenter) simbot.getSensorCenter(),"right",simbot, 5);
+//                    ColorIntensitySensor colorSensorR = new ColorIntensitySensor((SimBotSensorCenter) simbot.getSensorCenter(),"left",simbot, -5);
+//                    ColorIntensitySensor colorSensorM = new ColorIntensitySensor((SimBotSensorCenter) simbot.getSensorCenter(),"center",simbot, 0);
+                    BaseStation.getInstance().getBotManager().addBot(newBot);
+                }
+            }
             return addInfo;
         });
 
+        /**
+         * POST /saveScenario saves the scenario currently loaded as a txt
+         * file with the specified name
+         *
+         * @apiParam scenario a string representing a list of JSON scenario
+         * objects, which consist of obstacles and bot(s)
+         * @apiParam name the name the new scenario txt file
+         * @return the name of the file if it was successfully saved
+         */
+        post("/saveScenario", (req,res) -> {
+
+            String body = req.body();
+            JsonObject scenario = jsonParser.parse(body).getAsJsonObject();
+            String scenarioBody = scenario.get("scenario").getAsString();
+            String fileName = scenario.get("name").getAsString();
+
+            //writing new scenario file
+            File file = new File
+                    ("cs-minibot-platform-src/src/main/resources" +
+                            "/public/scenario/"+fileName+".txt");
+            OutputStream out = new FileOutputStream(file);
+
+            FileWriter writer = new FileWriter(file, false);
+            BufferedWriter bwriter = new BufferedWriter(writer);
+            bwriter.write(scenarioBody);
+            bwriter.close();
+            out.close();
+            return fileName;
+        });
+
+        /*loads a scenario with the specified file name to the simulator
```

Comment:
hi danny which part of the code does this floating comment belong to? Trevor would like it to be more obvious